src/Runner.Listener/MessageListener.cs

```diff
@@ -183,27 +199,52 @@ public async Task<TaskAgentMessage> GetNextMessageAsync(CancellationToken token)
             while (true)
             {
                 token.ThrowIfCancellationRequested();
+                jobStatusNotification = new TaskCompletionSource<bool>();
```

Comment:
Do we need to have both `jobStatusNotification` and `getMessagesTokenSource`?
will only have a global `_getMessagesTokenSource` work? let the OnJobStarted() to call `_getMessagesTokenSource?.cancel();`